Upgrade process from 4.0.x to 7.0.x (#863)
opencloud-eu/docs: docs/admin/maintenance/upgrade/upgrade-7.x.x.md

---
sidebar_position: 4
id: upgrade-7.x.x
title: Upgrade 7.x.x
description: Upgrading to OpenCloud 7.x.x
draft: false
---

import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'

# Upgrading to OpenCloud 7.x.x

This guide describes how to upgrade an existing OpenCloud deployment to OpenCloud 7.x.x.

The guide applies to upgrades from:

- OpenCloud 6.x rolling
- OpenCloud 4.x stable

It covers the required configuration migration, optional OpenSearch index rebuild, and web extension updates introduced with OpenCloud 7.x.x.

## Before You Start

> - Make sure your current OpenCloud deployment runs OpenCloud 6.x rolling or OpenCloud 4.x stable.

Reviewer: ```suggestion
- Make sure your current deployment is running on OpenCloud 6.x (rolling) or OpenCloud 4.x (stable).
```

- Create a complete backup of your configuration and data.
- Make sure you have access to your current `opencloud.yaml`.
- Verify that your current deployment is healthy before upgrading.
- Review the OpenCloud 7.x.x release notes before starting the upgrade.

## Back Up Configuration and Data

:::important

> Always create a backup before upgrading to prevent data loss.

Reviewer: You already wrote this in line 26

:::

<Tabs>
  <TabItem value="bind-mounts" label="Using Bind Mounts">

> If your configuration and data are stored in host directories, create a copy of these directories.

Reviewer: what are host directories?

Example:

```bash
cp -a /mnt/opencloud/config /mnt/opencloud/config-backup
cp -a /mnt/opencloud/data /mnt/opencloud/data-backup
```

Replace the paths with the directories used by your deployment.

  </TabItem>

  <TabItem value="named-volumes" label="Using Named Volumes">

Create a backup directory and copy the configuration and data from the OpenCloud container.

```bash
mkdir -p ~/opencloud-backups
docker cp opencloud-compose-opencloud-1:/etc/opencloud ~/opencloud-backups/config-backup
docker cp opencloud-compose-opencloud-1:/var/lib/opencloud ~/opencloud-backups/data-backup
```

Replace `opencloud-compose-opencloud-1` with the name of your OpenCloud container if it differs.

  </TabItem>
</Tabs>

## Update the `opencloud-compose` Checkout

> If you use the `opencloud-compose` repository, update your local checkout before pulling the new container image.

Reviewer: ```suggestion
If you use the `opencloud-compose` repository, update your local copy of the repository.
```

```bash
cd /opencloud-compose
git pull
```

> Skip this step if you run OpenCloud with plain Docker.

Reviewer: IMHO obsolete since this is obvious.

## Stop OpenCloud

Stop the currently running OpenCloud instance.

<Tabs>
  <TabItem value="docker-compose" label="Docker Compose">

```bash
docker compose stop
```

  </TabItem>

  <TabItem value="docker" label="Docker">

```bash
docker stop opencloud
```

  </TabItem>
</Tabs>

## Pull the OpenCloud 7.x.x Image

Pull the OpenCloud 7.x.x image.

```bash
docker pull opencloudeu/opencloud:7.x.x
```

> Replace `7.x.x` with the exact OpenCloud version you want to upgrade to.

Reviewer: Might need a description of the differences between the tags `:7` and `:7.x` or `7.x.x`

## Apply the Configuration Migration

OpenCloud 7.x.x includes a breaking configuration change that must be applied before starting the upgraded instance.

Reference: [Migration PR](https://github.com/opencloud-eu/opencloud/pull/2760)

### Open a Temporary OpenCloud Container

Open a shell in a temporary OpenCloud 7.x.x container and mount your OpenCloud configuration directory or configuration volume to `/etc/opencloud`.

<Tabs>
  <TabItem value="named-volumes" label="Using Named Volumes">

```bash
docker run --rm -it --entrypoint /bin/sh -v "<opencloud-config-volume>:/etc/opencloud" opencloudeu/opencloud:7.x.x
```

> Replace `<opencloud-config-volume>` with the Docker volume that contains your OpenCloud configuration.

Reviewer: this might not necessarily be a docker volume, this could also be a bind mount....

  </TabItem>
</Tabs>

TODO:
Release note 7.1.0 -> log sharing auf info stellen
Wenn die migration fertig ist, kommt ne info im log dass sie fertig ist
Am Anfang kann es etwas dauern shares gehen nicht und Mitglieder liste (was in den Release notes steht)

### ⏳ First Startup After Upgrading to OpenCloud 7.x.x

After upgrading to OpenCloud 7.x.x, the first startup may take several minutes before all Spaces become fully available.

During this period, some Space-related functionality is temporarily restricted:

- Space member lists may be incomplete or incorrect.
- Space memberships cannot be modified.
- Creating or deleting Spaces is not possible.
- Share-related operations may be unavailable until the migration has completed.

#### Background

> OpenCloud 7.x.x includes an updated Share Manager migration. Due to an issue in OpenCloud 7.0.0, some disabled Spaces were not fully processed during the initial migration. To ensure all existing Space memberships are migrated correctly, the migration task is executed again during the first startup after upgrading.

Reviewer 2: Please use a text from the 7.0.0 release notes as  a guidiance (https://github.com/opencloud-eu/opencloud/releases/tag/v7.0.0).

The current text seems to be based on the 7.1.0 release notes. People upgrading from 4.0.7 to the next stable release won't be affected by that issue. For them the stuff that is written in the 7.0.0  release notes about the spacemember migration is most relevant.

#### Background

> OpenCloud 7.0.0 introduced a new backend mechanism for managing Space memberships. When upgrading from an earlier release, the `sharing` service automatically migrates existing memberships to the new format.

Reviewer 2: I am not sure we should mention 7.0.0 here. People that just follow the stable releases won't even know that 7.0.0 ever existed. Maybe just leave it at 7.x

> In OpenCloud 7.1.0, the migration process was updated to address an issue where some disabled Spaces were not fully processed during the initial migration. To ensure all existing Space memberships are migrated correctly, the migration task is executed again during the first startup after upgrading.

Reviewer 2: You can leave out this paragraph. People updating from 4.0.x to the next stable release are not affected by this. This was purely an issue for people that were on 7.0.0 at some point.
